src/maestral/cli.py

```diff
@@ -1492,3 +1493,192 @@ def log_level(level_name: str, config_name: str) -> None:
         else:
             level_name = logging.getLevelName(m.log_level)
             cli.echo(f"Log level: {level_name}")
+
+
+@main.command(
+    section="Maintenance",
+    help="""
+Compare changes of two revisions of a file.
+If the second revision is omitted, it will compare the file to the current version.
+""",
+)
+@click.argument("dropbox_path", type=click.Path())
+@click.option(
+    "-v",
+    "--rev",
+    help="Revisions to compare (mulitple allowed to compare two old revs)",
+    multiple=True,
+    default=[],
+)
+@catch_maestral_errors
+@existing_config_option
+# If new_version_hash is omitted, use the current version of the file
+def diff(dropbox_path: str, rev: List[str], config_name: str) -> None:
+
+    import difflib
+    import tempfile
+    import magic
+    from datetime import datetime
+    from .daemon import MaestralProxy
+
+    # Some colors
+    RED = "\033[91m"
+    CYAN = "\033[96m"
+    GREEN = "\033[92m"
+    YELLOW = "\033[93m"
+    RESET = "\033[0m"
+    BOLD = "\033[1m"
+
+    # Reason for rel_dbx_path: os.path.join does not like leading /
+    if not dropbox_path.startswith("/"):
+        dropbox_path = "/" + dropbox_path
+
+    def download_and_compare(m: MaestralProxy, old_rev: str, new_rev: str = None):
+        """
+        Download up to two revisions to a local temporary folder
+        and compare them with a 'diff'. Only text files are supported.
+        If an unknown file type was found, everything that doesn't match
+        'text/*', an error message gets printed.
+        """
+        # Create a temporary directory to store all downloaded files
+        tempdir = tempfile.TemporaryDirectory()
+
+        # Get the dates of the revisions
+        entries = m.list_revisions(dropbox_path)
+        new_date = check_rev(entries, entries[0]["rev"])
+        if new_date is None:
+            click.echo("Selected revision was not found")
+            return
+        old_date = check_rev(entries, old_rev)
+        if old_date is None:
+            click.echo("Selected revision was not found")
+            return
+
+        new_location = os.path.join(m.dropbox_path, dropbox_path[1:])
+        old_location = os.path.join(tempdir.name, old_rev)
+
+        # Download specific revision to cache
+        _ = m.download_revision(dropbox_path, old_location, old_rev)
+
+        # Use the current version if new_version_hash is None
+        # Saves space (unnecessary downloads omitted)
+        if new_rev:
+            new_date = check_rev(entries, new_rev)
+            if new_date is None:
+                click.echo("Selected revision was not found")
+                return
+            new_location = os.path.join(tempdir.name, new_rev)
+            _ = m.download_revision(dropbox_path, new_location, new_rev)
+
+        with open(new_location) as f:
+            new_content = f.readlines()
+        with open(old_location) as f:
+            old_content = f.readlines()
```

Comment:
Should this be done in the `get_file_diff` I craeted in the `Maestral` class or on the cli side? I guess on the cli side
.

Reply:
Good question. I can see the following scenarios:
 
1) An error where the file no longer exists is only likely when taking the current rev from the local Dropbox folder. The user is unlikely to mess with the actual temp dir (but this may still happen). The best way to handle either would be to fall back to downloading the rev from Dropbox servers instead of raising an error.

2) An `UnicodeDecodeError` will be more common. We should probably convert this to a `maestral.errors.UnsupportedFileError` with an understandable error message.

3) Other OSErrors should be converted to the appropriate `MaestralApiError` using [os_to_maestral_error](https://github.com/SamSchott/maestral/blob/e9e6237cb264a3cac926921cbb646804d26e28c4/src/maestral/client.py#L1054) and then reraised. This can be done using the convenient context manager [convert_api_errors](https://github.com/SamSchott/maestral/blob/e9e6237cb264a3cac926921cbb646804d26e28c4/src/maestral/client.py#L173) and will cover all sorts of exotic errors (too little disk space, out-of-memory, etc) while providing an actionable message to the user.

Reply:
Those `MaestralAPIError`s can then be raised by the API and will cause the CLI to fall back to its default error handling: aborting and printing the error message to the console.

Reply:
Should we abort after a second download/reading failed? Or do you mean it should retry infinetely?

Reply:
No, retrying infinitely would be a bad idea... Thinking about it, it's probably not even necessary to retry downloads to the temp dir. But falling back to a download when the latest rev disappeared from the local Dropbox folder makes sense. This is a comparatively likely scenario.

Reply:
Instead of waiting for the error I just checked in the beginning if the file exists with `os.path.exists(...)`.
It seems I fixed all errors, let me know if I overlooked something.

Reply:
It's typically always better in such cases to try to open file file and handle errors as they occur instead of checking in advance. This eliminates the possibility of the file being deleted after the check but before opening.

Reply:
Ok, I'll have another detailed look at the code tomorrow. Thanks for making all the changes!